src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -641,6 +672,12 @@ public MetadataEnumeration<DatasetSummary> enumerate(
     // add the filters to the clause to get the actual items
     DaoUtils.addFilterClause(filter, params, whereClauses, TABLE_NAME);
     DaoUtils.addRegionFilterClause(region, params, whereClauses, "dataset.id");
+    try (Connection connection = jdbcDataSource.getConnection()) {
+      TagUtils.addTagsClause(connection, tags, params, whereClauses, TABLE_NAME);
+    } catch (SQLException e) {
+      throw new IllegalArgumentException(
+          "Failed to convert dataset request tags list to SQL array", e);
+    }
```

Comment:
This is totally valid so just musing - I think I would have pushed this try/catch logic into the tag utils method. Is there a reason why you didn't/why that wouldn't work?

Reply:
Good question! I wanted to keep the TagUtils method static, and this requires that we get our live connection from an instance variable.